src/claude-session-utility.ts

```
@@ -1132,19 +1204,50 @@ const detectActiveSessionsLegacy = async (activeMap: Map<string, number>): Promi
+ * launcher interpolates the path into a double-quoted shell string and, for
+ * Ghostty and Terminal.app, into an AppleScript string literal; a path
+ * carrying a quote, a backslash, `$`, a backtick or a line break could end
+ * either literal. Real project paths never contain these, so refusing them
+ * costs nothing and closes the hole for every caller — the saved-list file
+ * and history.jsonl are both plain files a user (or a stray tool) can edit.
+ * An EMPTY path is not this function's concern: switching to a running
+ * session never embeds it (iTerm2 matches by tty), and the launch paths that
+ * do need one already check existence.
+ */
+export const isSafeLaunchPath = (p: unknown): p is string =>
+  typeof p === 'string' && !/["\\$`\n\r\u0000]/.test(p);
```

ESLint

[error] Unexpected control character(s) in regular expression: \x00.

(no-control-regex)